lib/msf/core/auxiliary/mikrotik.rb

```diff
@@ -185,11 +185,11 @@ def mikrotik_routeros_config_eater(thost, tport, config)
           module_fullname: fullname,
           status: Metasploit::Model::Login::Status::UNTRIED
         }
-      end
 
-      # Default SNMP to UDP
-      if tport == 161
-        credential_data[:protocol] = 'udp'
+        # Default SNMP to UDP
+        if tport == 161
+          credential_data[:protocol] = 'udp'
```

Comment:
When `framework.db.active` is false - it causes `credential_data` to be nil - then this crashes